fix(mr): Readd ability to close multiple MRs

zemzale opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

**Description**
Ability to remove multiple MRs was there but it had gone missing when
implementing `MRFromArgs`.

Added a function for concurrently fetching multiple merge requests from
branches that, falsback to the `MRFromArgs` if the argument count is
lower then 1. This is because the MRFromArgs, contains some specific
logic for handling cases where there are no arguments or user provided
0 in arguments.

Added this functionality to deal with more then one MR to `delete` and `reopen`. 
I opted to not implement this functionality to `merge` command since, if there merge multiple MRs one might create merge conflicts to others and block the merging.  

**Related Issue**
Fixes #301


**Types of changes**
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

zemzale committed:
fix(mr): Readd ability to close multiple MRs

Ability to remove multiple MRs was there but it had gone missing when
implementing `MRFromArgs`.

Added a function for concurrently fetching multiple merge requests from
branches that, fallsback to the `MRFromArgs` if the argument count is
lower then 1. This is because the MRFromArgs, contains some specific
logic for handling cases where there are no arguments or user provided
0 in arguments.

Fixes #301

zemzale committed:
fix(mr): Readd missing ability to reopen multiple MRs

There was ability to have reopen more then one MR. This was lost when
`mrutils.MRsFromArgs` was implemented.

zemzale committed:
fix(mr): Readd missing ability to delete multiple MRs

profclems commented on commands/mr/mrutils/mrutils.go:
```diff
@@ -151,6 +153,54 @@ func MRFromArgs(f *cmdutils.Factory, args []string) (*gitlab.MergeRequest, glrep
 }
 
+func MRsFromArgs(f *cmdutils.Factory, args []string) ([]*gitlab.MergeRequest, glrepo.Interface, error) {
+	if len(args) <= 1 {
```
There is a restriction on the `mr close` command on this line indicating it cannot have more than one positional arguments:
https://github.com/profclems/glab/blob/9b0868068e88acdec171aa07d6651df537b72986/commands/mr/close/mr_close.go#L21

I'm glad you did it this way so we can have room for cases where we allow multiple positional arguments (`glab mr close 1 2 3 4`) but for backward compatibility where the user can do `glab mr close 1,2,3,4` to close multiple MRs , you should check for when there is a single argument with comma-separated IDs.

```go
if len(args) <= 1 {
        var arrIds []string
        if(len(args) == 1) {
	        arrIds := strings.Split(mergeID, ",")
        }
        if (len(arrids) <= 1) {
		mrID, baseRepo, err := MRFromArgs(f, args)
		if err != nil {
			return nil, nil, err
		}
		return []*gitlab.MergeRequest{mrID}, baseRepo, err
	}
        args = arrIds
}
```

zemzale replied:
I didn't think of the backwards compatibility with comma separated arguments. 
In this MR I ended up removing the constrain to the maximum arguments, but the correct thing would be to just revert back to the single argument and have it comma separated, and do it this way.

zemzale replied:
Off-topic: Do you mind that I force push to my PRs? 
I know some people don't like when someone does that because they like to go through PRs commit by commit, but I like to have my commits tidy, and if I make a mistake in a commit I ammend to it and force push it, like the mistake never existed.

profclems replied:
That's okay

profclems replied:
You can merge this PR now. I have reviewed and approved

zemzale committed:
fix(mrutils): Allow for comma separated MR arguments

For backwards compatibility reasons, we gotta allow single, comma
separated argument passed to the `MRsFromArgs` function.

zemzale committed:
fix(mr): Allow only one argument to mr commands
